rust_src/src/coding.rs

```diff
@@ -98,4 +99,31 @@ pub fn encode_file_name(fname: LispStringRef) -> LispStringRef {
     unsafe { c_encode_file_name(fname.into()) }.into()
 }
 
+/// Decode STRING which is encoded in CODING-SYSTEM, and return the result.
+/// Optional third arg NOCOPY non-nil means it is OK to return STRING
+/// itself if the decoding operation is trivial.
+/// Optional fourth arg BUFFER non-nil means that the decoded text is inserted in that buffer after point (point does not move). In this case, the return value is the length of the decoded text.
+/// This function sets `last-coding-system-used` to the precise coding system
+/// used (which may be different from CODING-SYSTEM if CODING-SYSTEM is not fully specified.)
+#[lisp_fn(min = "1")]
+pub fn decode_coding_string(string: LispObject, coding_system: LispObject, nocopy: Option<LispObject>, buffer: Option<LispObject>) -> LispObject {
+    code_convert_string(string, coding_system, buffer, false, nocopy.is_some(), false)
+}
+
+/// Encode STRING to CODING-SYSTEM, and return the result.
+/// Optional third arg NOCOPY non-nil means it is OK to return STRING
+/// itself if the encoding operation is trivial.
+/// Optional fourth arg BUFFER non-nil means that the encoded text is inserted in that buffer after point (point does not move). In this case, the return value is the length of the encoded text.
+/// This function sets `last-coding-system-used` to the precise coding system
+/// used (which may be different from CODING-SYSTEM if CODING-SYSTEM is not fully specified.)
+#[lisp_fn(min = "1")]
+pub fn encode_coding_string(string: LispObject, coding_system: LispObject, nocopy: Option<LispObject>, buffer: Option<LispObject>) -> LispObject {
+    code_convert_string(string, coding_system, buffer, true, nocopy.is_some(), false)
+}
+
+// Wrapper for code_convert_string (NOT PORTED)
+pub fn code_convert_string(string: LispObject, coding_system: LispObject, dst_object: Option<LispObject>, encodep: bool, nocopy: bool, norecord: bool) -> LispObject {
+    unsafe { c_code_convert_string(string.into(), coding_system.into(), dst_object.into(), encodep.into(), nocopy.into(), norecord.into()).into() }
```

Comment:
`into` is only need to cast from one type to another. It turns A *into* B. Since all of your types are already `LispObject` there is no need to transform them.